romancal/assign_wcs/assign_wcs_step.py

```diff
@@ -0,0 +1,73 @@
+"""
+Assign a gWCS object to a science image.
+
+"""
+import logging
+
+from astropy import coordinates as coord
+from astropy import units as u
+import gwcs.coordinate_frames as cf
+
+from gwcs.wcs import WCS, Step
+
+from roman_datamodels import datamodels as rdm
+from romancal.stpipe import RomanStep
+from . import pointing
+from .utils import wcs_bbox_from_shape
+
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+__all__ = ["AssignWcsStep", "load_wcs"]
+
+
+class AssignWcsStep(RomanStep):
+    """ Assign a gWCS object to a science image.
+    """
+
+    reference_file_types = []
+
+    def process(self, input_model):
+        reference_file_names = {}
+        with rdm.open(input_model) as input_model:
+            for reftype in self.reference_file_types:
+                log.info(f'reftype, {reftype}')
+                reffile = self.get_reference_file(input_model, reftype)
+                reference_file_names[reftype] = reffile if reffile else ""
+            log.debug(f'reference files used in assign_wcs: {reference_file_names}')
+            input_model = self.open_model(input_model)
+            result = load_wcs(input_model.copy(), reference_file_names)
+        return result
+
+
+def load_wcs(dmodel, reference_files):
+    """ Create a gWCS object and store it in ``Model.meta``.
+
+    Parameters
+    ----------
+    dmodel : `~roman_datamodels.datamodels.WfiImage`
+        The exposure.
+    reference_files : dict
+        A dict {reftype: reference_file_name} containing all
+        reference files that apply to this exposure.
+    """
```

Comment:
Could include `return` section in epydoc for updated dmodel.

Reply:
done